### PR TITLE
[CorceBundle] Add sylius_order_state_choice form type

### PR DESCRIPTION
**Pull request by aramalipoor**
Add `sylius_order_state_choice` form type and also simplify `sylius_advanced_order` fields.

### CODE REVIEW

**Comment by pjedrzejewski** [comment]
:+1:

**Comment by stloyd** [comment]
I would prefer to put this form in OrderBundle instead as it don't depend on Core features.

**Comment by pjedrzejewski** [comment]
Agree with @stloyd!

File: src/Sylius/Bundle/CoreBundle/Form/Type/OrderStateChoiceType.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+*
+* For the full copyright and license information, please view the LICENSE
+* file that was distributed with this source code.
+*/
+
+namespace Sylius\Bundle\CoreBundle\Form\Type;
+
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\OptionsResolver\OptionsResolverInterface;
+use Sylius\Component\Core\Model\OrderInterface;
```
Comment (stloyd) [inline src/Sylius/Bundle/CoreBundle/Form/Type/OrderStateChoiceType.php]: After move suggested in comments this should be changed to `Order` namespace.

Comment (aramalipoor) [inline src/Sylius/Bundle/CoreBundle/Form/Type/OrderStateChoiceType.php]: @stloyd Oh, thanks for the catch! :D

**Comment by aramalipoor** [comment]
@pjedrzejewski Done! :)

**Comment by aramalipoor** [comment]
@pjedrzejewski I don't think the fail is related to this change! Would you please check it out?

**Comment by pjedrzejewski** [comment]
Restarted the failing build. :)

**Comment by aramalipoor** [comment]
@pjedrzejewski Good to go.

**Comment by pjedrzejewski** [comment]
Thanks a lot Aram! :+1:

**Pull request merged**